File: BackEnd/Application/UseCases/Projects/GetTicketStatusesByProjectId/GetTicketStatusesByProjectIdQueryHandler.cs
using Domain.Aggregates;
using Infrastructure.Repository;
using MediatR;
using Microsoft.EntityFrameworkCore;

namespace Application.UseCases.Projects.GetTicketStatusesByProjectId;

public class GetTicketStatusesByProjectIdQueryHandler(IRepository<TicketStatus, Guid> ticketStatusRepository) : IRequestHandler<GetTicketStatusesByProjectIdQuery, List<TicketStatus>>
{
    public async Task<List<TicketStatus>> Handle(GetTicketStatusesByProjectIdQuery request, CancellationToken cancellationToken)
    {
        var ticketStatuses = await ticketStatusRepository.Where(x => x.ProjectId.Equals(request.ProjectId))
            .ToListAsync(cancellationToken);
        return ticketStatuses;
    }
}

File: flutter/lib/application/use_cases/projects/get_ticket_statuses/get_ticket_statuses_by_project_id_query_handler.dart
import 'package:enterprise_management/application/use_cases/projects/get_ticket_statuses/get_ticket_statuses_by_project_id_query.dart';
import 'package:enterprise_management/domain/aggregates/ticket_status/ticket_status.dart';
import 'package:enterprise_management/infrastructure/repositories/project_repository.dart';
import 'package:enterprise_management/shared_kernel/cqrs/cqrs.dart';
import 'package:enterprise_management/shared_kernel/result/result.dart';

class GetTicketStatusesByProjectIdQueryHandler extends IQueryHandler<GetTicketStatusesByProjectIdQuery, List<TicketStatus>> {
  const GetTicketStatusesByProjectIdQueryHandler({required this._projectRepository});
  final IProjectRepository _projectRepository;

  @override
  Future<Result<List<TicketStatus>>> handle(GetTicketStatusesByProjectIdQuery query) async {
    final res = await _projectRepository.getTicketStatusesByProjectId(projectId: query.projectId);
    return Result.success(res);
  }
}